tests/testTransforms.py

```diff
@@ -0,0 +1,468 @@
+        numpy.random.seed(50)
+
+    def makeRandom(self):
+        return makeRandomPolynomialTransform(4)
+
+    def testArrayConstructor(self):
+        """Test that construction with coefficient arrays yields an object with
+        copies of those arrays, and that all dimensions must be the same.
+        """
+        order = 3
+        xc = makeRandomCoefficientMatrix(order + 1)
+        yc = makeRandomCoefficientMatrix(order + 1)
+        p = PolynomialTransform(xc, yc)
+        self.assertEqual(p.getOrder(), order)
+        self.assertClose(p.getXCoeffs(), xc, atol=0, rtol=0)
+        self.assertClose(p.getYCoeffs(), yc, atol=0, rtol=0)
+        # Test that the coefficients are not a view.
+        xc[0, 0] = 50.0
+        self.assertNotEqual(p.getXCoeffs()[0, 0], 50.0)
+        # Test that rectangular coefficient arrays are not allowed.
+        self.assertRaises(
+            lsst.pex.exceptions.LengthError,
+            PolynomialTransform,
+            numpy.zeros((5, 4), dtype=float),
+            numpy.zeros((5, 4), dtype=float)
+        )
+        # Test that x and y coefficient arrays must have the same shape.
+        self.assertRaises(
+            lsst.pex.exceptions.LengthError,
+            PolynomialTransform,
+            numpy.zeros((5, 5), dtype=float),
+            numpy.zeros((4, 4), dtype=float)
+        )
+
+    def testConvertScaledPolynomial(self):
+        """Test that we can convert a ScaledPolynomialTransform to a PolynomialTransform.
+        """
+        scaled = makeRandomScaledPolynomialTransform(4)
+        converted = PolynomialTransform.convert(scaled)
+        self.checkTransformEquivalence(scaled, converted)
+
+    def testConvertSipForward(self):
+        """Test that we can convert a SipForwardTransform to a PolynomialTransform.
+        """
+        sipForward = makeRandomSipForwardTransform(4)
+        converted = PolynomialTransform.convert(sipForward)
+        self.checkTransformEquivalence(sipForward, converted)
+
+    def testConvertSipReverse(self):
+        """Test that we can convert a SipForwardTransform to a PolynomialTransform.
+        """
+        sipReverse = makeRandomSipReverseTransform(4)
+        converted = PolynomialTransform.convert(sipReverse)
+        self.checkTransformEquivalence(sipReverse, converted)
+
+    def testCompose(self):
+        """Test that AffineTransforms and PolynomialTransforms can be composed
+        into an equivalent PolynomialTransform.
+        """
+        poly = makeRandomPolynomialTransform(4)
+        affine = lsst.afw.geom.AffineTransform(
+            lsst.afw.geom.LinearTransform(numpy.random.randn(2,2)),
+            lsst.afw.geom.Extent2D(*numpy.random.randn(2))
+        )
+        composed1 = lsst.meas.astrom.compose(poly, affine)
+        composed2 = lsst.meas.astrom.compose(affine, poly)
+        self.checkTransformEquivalence(composed1, lambda p: poly(affine(p)))
+        self.checkTransformEquivalence(composed2, lambda p: affine(poly(p)))
```

Comment:
The test doesn't quite seem to do what is advertised: check that the result of composing gives the expected result. I suggest you define a python transform that applies affine and then poly (or vice-versa) and check equivalence (similar to what `ScaledPolynomialTransformTestCase .testConstruction` does below)